feat(auth): implement login/signup as popups

SayliDev opened the pull request:
## Description
Cette Pull Request **réorganise l'authentification** dans la marketplace en **déplaçant les pages de connexion et d'inscription sous forme de popups**. L'objectif est de **permettre aux utilisateurs de s’inscrire et de se connecter via popup** plutôt que de rediriger vers des pages séparées.

---

> [!IMPORTANT]
> Avant de tester, ajouter cette variable d’environnement dans le fichier **`.env`** du dossier **client** :  
>```env
>VITE_API_BASE_URL=http://localhost:8080
>```

## Fonctionnalités développées  

### 🏗️ **Refactorisation de l'authentification en Popup**  

💡 **Détails des fonctionnalités** :  

✅ **Déplacement des pages Login/Signup vers des popups**  
- Les formulaires de connexion et d'inscription sont désormais intégrés dans des popups.  
- Navigation entre les pages Login et Signup directement dans le popup.  
- Fermeture automatique du popup après une inscription/connexion réussie.  

✅ **Comportement du popup**  
- Le popup d'authentification s'ouvre au clic sur l'icône utilisateur dans le header.  
- Empêche l'interaction avec l'overlay une fois le popup fermé.  

✅ **Gestion des erreurs et validation**  
- Validation des champs du formulaire (email valide, mot de passe sécurisé, etc.).  
- Affichage des messages d'erreur pour les erreurs d'authentification.  

✅ **Comportement mobile amélioré**  
- Amélioration de la mise en page mobile.  

---

## 🧪 **Tests E2E avec Cypress**  
📌 **Fichiers concernés** :  
- `cypress/e2e/auth.spec.cy.ts`  

🛠️ **Liste des tests réalisés** :  
🔹 Tests ajustés pour vérifier le bon comportement de la navigation dans les popups (connexion/inscription).  
🔹 Vérification du bon fonctionnement de la fermeture du popup après une action réussie.  

---

## 🔗 **Issue liée**  
- **Closes:** #101
- **Related to:** #98

0.15.2

SayliDev committed:
feat(auth): add base popup component

- Implemented reusable BasePopup component
- Used for authentication modals (login/signup)
- Supports open/close state management

SayliDev committed:
feat(auth): add base popup component

- Implemented reusable BasePopup component
- Used for authentication modals (login/signup)
- Supports open/close state management

SayliDev committed:
fix(auth): prevent MainLayout from re-rendering in AuthPopup

- Removed MainLayout from AuthLayout to avoid unnecessary re-renders
- Fixed performance issue causing Firefox to freeze when opening the popup

SayliDev committed:
fix(auth): prevent MainLayout from re-rendering in AuthPopup

- Removed MainLayout from AuthLayout to avoid unnecessary re-renders
- Fixed performance issue causing Firefox to freeze when opening the popup

SayliDev committed:
feat(auth): add variant prop to BasePopup for custom styling

SayliDev committed:
feat(auth): add variant prop to BasePopup for custom styling

SayliDev committed:
style(auth): improve responsive design for mobile

SayliDev committed:
style(auth): improve responsive design for mobile

SayliDev committed:
style(auth-popup): refine mobile layout for better usability

SayliDev committed:
style(auth-popup): refine mobile layout for better usability

SayliDev committed:
feat(auth)): add functionality to switch between login and signup pages

- Pass `setIsLogin` function as prop to `LoginPage` and `SignupPage`
- Allow navigation between login and signup forms within the popup

SayliDev committed:
feat(auth)): add functionality to switch between login and signup pages

- Pass `setIsLogin` function as prop to `LoginPage` and `SignupPage`
- Allow navigation between login and signup forms within the popup

SayliDev committed:
refactor(auth): move all auth-related types to a dedicated "Auth.ts" file

SayliDev committed:
refactor(auth): move all auth-related types to a dedicated "Auth.ts" file

SayliDev committed:
chore(tests): temporarily disable all Auth tests

SayliDev committed:
chore(tests): temporarily disable all Auth tests

SayliDev committed:
fix(location-popup): remove required children prop from LocationPopup props

- Created a dedicated LocationPopupProps interface that omits the 'children' prop
- LocationPopup now provides its own children to BasePopup, avoiding the TS error

SayliDev committed:
fix(location-popup): remove required children prop from LocationPopup props

- Created a dedicated LocationPopupProps interface that omits the 'children' prop
- LocationPopup now provides its own children to BasePopup, avoiding the TS error

SayliDev committed:
feat(auth): elevate login and signup routes into popup

SayliDev committed:
feat(auth): elevate login and signup routes into popup

SayliDev committed:
feat(auth): close popup automatically after login/signup

SayliDev committed:
feat(auth): close popup automatically after login/signup

SayliDev committed:
Merge branch '101-auth-popup' of https://github.com/Pinqo-community/marketplace-app into 101-auth-popup

SayliDev committed:
chore(auth): remove unnecessary redirect to login page

SayliDev committed:
fix(auth): prevent overlay blocking interaction after closing popup

SayliDev committed:
test: adjust popup navigation tests for login/signup flow

SayliDev committed:
refactor(auth): move LoginPage and SignupPage to components/Auth as popups

SayliDev committed:
chore: bump version to 0.15.0

Lisadev84 commented on client/src/components/Auth/SocialButtons.tsx:
```diff
@@ -0,0 +1,33 @@
+export const SocialButtons: React.FC<SocialButtonsProps> = ({ type }) => {
+  const googleAuthUrl = `${import.meta.env.VITE_API_BASE_URL}/api/v1/oauth2/authorization/google?redirect_uri=${window.location.origin}/oauth/redirect`;
+  const facebookAuthUrl = `${import.meta.env.VITE_API_BASE_URL}/api/v1/oauth2/authorization/facebook?redirect_uri=${window.location.origin}}/oauth/redirect`;
+
```
Une erreur apparaît à la connexion via facebook, je pense que tu as une accolade de trop dans la variable qui du coup apparait dans l'uri : 
`localhost:8080/api/v1/oauth2/authorization/facebook?redirect_uri=http://localhost:5173}/oauth/redirect`

SayliDev replied:
En effet j'avais pas fait attention... C'est corrigé, merci !

Lisadev84 reviewed (commented):
Une fois l'erreur dans l'uri corrigée, inscription et connexion sont opérationnelles classique ou Oauth. En revanche, les fichiers auth.spec.cy.ts et locationPopup.spec.cy.ts du répertoire cypress sont en erreur
`File is not included in any tsconfig.json`

SayliDev committed:
fix(auth): remove extra bracket in redirect Facebook URI

SayliDev committed:
chore: Bump dependencies to latest versions

SayliDev committed:
chore: fix vulnerabilities with npm audit fix

SayliDev commented:
> Une fois l'erreur dans l'uri corrigée, inscription et connexion sont opérationnelles classique ou Oauth. En revanche, les fichiers auth.spec.cy.ts et locationPopup.spec.cy.ts du répertoire cypress sont en erreur `File is not included in any tsconfig.json`

De mon côté, je vois pas d'erreur sur les fichiers Cypress. J’ai essayé de corriger potentiellement l’erreur en ajoutant le dossier cypress dans tsconfig.app.json, dis-moi si ça marche pour toi. Sinon, un `npm run dev` après un clean `(rm -rf node_modules/.cache)` peut aider

SayliDev committed:
chore(config): include cypress in tsconfig.app.json

SayliDev committed:
fix: add Cypress type to tsconfig.app.json

JeffreyGbeho commented on client/src/api/api.ts:
J'ai un problème d'url vers le back. Ca me donne ça par exemple pour google 
http://localhost:5173/undefined/api/v1/oauth2/authorization/google?redirect_uri=http://localhost:5173/oauth/redirect

Y'a un undefined juste après l'origine et l'origine devrait être localhost:8080 pour le back

SayliDev replied:
> J'ai un problème d'url vers le back. Ca me donne ça par exemple pour google http://localhost:5173/undefined/api/v1/oauth2/authorization/google?redirect_uri=http://localhost:5173/oauth/redirect
> 
> Y'a un undefined juste après l'origine et l'origine devrait être localhost:8080 pour le back

De mon côté, j’ai pas ce souci, l’URL est bien formée et je peux me connecter avec Google sans problème. T’as bien mis `VITE_API_BASE_URL=http://localhost:8080` (avec le bon port) dans ton `.env` du dossier client ? Et bien redémarré le serveur après ?

JeffreyGbeho replied:
Ah oui en effet j'avais oublié la variable d'environnement

JeffreyGbeho commented on client/src/components/Auth/Login.tsx:
```diff
@@ -0,0 +1,94 @@
+    try {
+      await login(dispatch, { email: trimmedEmail, password: trimmedPassword });
+      dispatch(closeAuthPopup());
+      console.log("Connexion réussie ✅");
```
Supprimer les console.log

JeffreyGbeho commented on client/src/components/Auth/OAuthRedirect.tsx:
```diff
@@ -0,0 +1,60 @@
+        }),
+      );
+      navigate("/");
+      console.log("Authentication reussie ✅", accessToken, refreshToken);
```
Supprimer le console log

JeffreyGbeho commented on client/src/components/Auth/Signup.tsx:
```diff
@@ -0,0 +1,178 @@
+    if (Object.keys(newErrors).length === 0) {
+      try {
+        setLoading(true);
+        console.log("Données envoyées:", formData);
```
Supprimer le console.log

JeffreyGbeho commented on client/src/components/Auth/Signup.tsx:
```diff
@@ -0,0 +1,178 @@
+          password: formData.password,
+        });
+        dispatch(closeAuthPopup());
+        console.log("Inscription réussie ✅");
```
Supprimer le console.log

JeffreyGbeho commented on client/src/components/Auth/Signup.tsx:
```diff
@@ -0,0 +1,178 @@
+      } catch (err) {
+        const error = err as AxiosError<{ message?: string }>;
+        console.error("Erreur lors de l'inscription", error);
+        console.log("Réponse de l'API:", error.response?.data);
```
Supprimer le console.log

JeffreyGbeho commented on client/src/components/Auth/AuthPopup.tsx:
Empêcher l'accès à la modale de connexion quand l'utilisateur est déjà connecté. Il doit se déconnecter pour pouvoir y avoir et si son token n'est pas valide et que il peut pas le refresh ça doit le déconnecter aussi.
Peut être modifier l'icone user quand il est connecté pour que l'utilisateur sache en un coup d'oeil si il est ou pas

SayliDev committed:
fix: add Cypress types to tsconfig.cypress.json for E2E tests

SayliDev committed:
feat(auth): toggle auth popup state based on access token presence

SayliDev committed:
feat(auth): allow user to log out from the UserButton component

- Prevents opening the auth popup if the user is already logged in
- Enables user logout when clicking the UserButton if authenticated

SayliDev committed:
chore(logging): remove sensitive user information from console logs

SayliDev committed:
chore: bump to 0.15.2

Lisadev84 reviewed (approved):
Les erreurs sur le fichiers .ts ne se produisent plus. 
Le bouton déconnection est opérationnel et permet d'office à l'utilisateur de connaître son satut. 
Good Job ;.)

SayliDev committed:
Merge branch 'develop' into 101-auth-popup
